android/src/main/java/com/swmansion/reanimated/NodesManager.java

```diff
@@ -427,29 +430,44 @@ public void onEventDispatch(Event event) {
     // Events can be dispatched from any thread so we have to make sure handleEvent is run from the
     // UI thread.
     if (UiThreadUtil.isOnUiThread()) {
-      handleEvent(event);
+      handleEvent(event, false);
     } else {
-      mEventQueue.offer(event);
+      synchronized (mCopyingEventEmitter) {
```

Comment:
why is this here? does not seem to be needed, we don't synchronize on mCopyingEventEmitter elsewhere and the code that emitter runs does not seem to be not thread safe?

Reply:
If I remember correctly I wanted to synchronize `mEventMapping.contains` key check with adding to the event queue, so we don't remove mapping while the event is copied and added to the queue. 
It may be unnecessary here because we queue events anyway and we should check if a mapping exists at a time when we're popping event from the queue.

Reply:
if eventmapping can be modified in a different thread there should be some level of synchronization. But synchronizing on it in only one place is not enough. I see you changed the map to mEventMapping to a concurrent map. This should address the problem of the map internals when dealing with reads writs from multiple threads. IMO this seems enough and the synchronized block isn't needed. If you want to ensure that the mapping stays in the map until we process the event we'd need to add some additional locking, but I don't think it is necessary. The map check is only added as an optimization step so that we don't copy all the events but only ones we want to process.